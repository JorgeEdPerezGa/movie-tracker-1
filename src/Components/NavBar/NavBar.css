.nav-bar {
  position: fixed;
  display: flex;
  justify-content: space-around;
  width: 100%;
  background: black;
  padding-top: 30px;
  height: 50px; }

.welcome {
  color: white; }

.nav {
  color: white; }

.logo {
  color: blue;
  margin: 0px; }

.logo-two {
  color: white; }
