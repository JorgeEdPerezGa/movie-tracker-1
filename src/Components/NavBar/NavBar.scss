.nav-bar {
  z-index: 1;
  position: fixed;
  display: flex;
  justify-content: space-around;
  width: 100%;
  background: black;
  padding-top: 30px;
  height: 50px;
}

.welcome{
  // font-weight: 100;
  font-family: 'Comfortaa', cursive;
  display: block;
  color: #817df1;
  margin: 0;
}

.nav {
  color: white;
  text-decoration-line: none;
  &:focus {
    color: #817df1;
  }
}

.logout {
  font-weight: 100;
  border: none;
  background: #817df1;
  color: #ffffff;
  height: 22px;
  width: 110px;
  border-radius: 100px;
}

.logo{
  font-family: 'Comfortaa', cursive;
  color: #817df1;
  margin: 0px;
}

.logo-two {
  font-family: 'Comfortaa', cursive;
  color: white
}

.user {
  font-family: 'Comfortaa', cursive;
  color: #817df1;
}
