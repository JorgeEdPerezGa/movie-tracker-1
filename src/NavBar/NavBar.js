import React from 'react';
import { NavLink } from 'react-router-dom';

const NavBar = () => {
  return (
    <div className='nav-bar'>
      <h1>JORGES NAV BAR</h1>
      <NavLink className='nav' to='/movies'>JORGES Movies</NavLink>
    </div>
  )
}

export default NavBar;
