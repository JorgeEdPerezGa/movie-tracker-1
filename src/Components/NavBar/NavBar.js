import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { logout } from '../../Actions/Actions';
import './NavBar.css'
import helper from '../../API/helper.js';
import { addMovies } from '../../Actions/Actions';
const { fetchMovies } = helper;

class NavBar extends Component {


 async componentDidMount() {
    const getMovies = await fetchMovies();
    const movieData = this.props.addMovieFunction(getMovies);
}

render() {
  if(Object.keys(this.props.user).length === 0) {
  return (
    <div className='nav-bar'>
      <p className='logo'>MOVIE <span className = 'logo-two'>TRACKER</span></p>
      <NavLink className='nav' to='/'>Movies</NavLink>
      <NavLink className='nav' to='/login'>Login</NavLink>
      <NavLink className='nav' to='/register'>Register</NavLink>
    </div>
  )
}
 else {
  return (
    <div className='nav-bar'>
      <NavLink className='nav' to='/'>Movies</NavLink>
      <h4>Welcome</h4>
      <button onClick = {this.props.logout}>log out</button>
    </div>
  )
 }
}
}


export const mapStateToProps = (store) => {
  return {
    user: store.user
  }
}

export const mapDispatchToProps = (dispatch) => {
  return {
    logout: () => {
      dispatch(logout())
    },
     addMovieFunction: (getMovies) => {
    dispatch(addMovies(getMovies));
   }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(NavBar)
