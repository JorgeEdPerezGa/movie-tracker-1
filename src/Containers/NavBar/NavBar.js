import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { logout, clearFavorites } from '../../Actions/Actions';
import './NavBar.css'
import helper from '../../API/helper.js';
import { addMovies } from '../../Actions/Actions';
import PropTypes from 'prop-types';

const { fetchMovies } = helper;

export class NavBar extends Component {


 async componentDidMount() {
    const getMovies = await fetchMovies();
    const movieData = this.props.addMovieFunction(getMovies);
}

logoutRedirect = (props) => {
  this.props.logout()
  this.props.clearFavorites();
  this.props.history.push('/login');
}

render() {
  const favoriteLength = this.props.favorites.length;

  if(Object.keys(this.props.user).length === 0) {
  return (
    <div className='nav-bar'>
      <p className='logo'>MOVIE <span className = 'logo-two'>TRACKER</span></p>
      <NavLink className='nav' to='/'>Movies</NavLink>
      <NavLink className='nav' to='/login'>Login</NavLink>
      <NavLink className='nav' to='/register'>Register</NavLink>
    </div>
  )
}
 else {
   let mappedName = this.props.user.map(name => name.data.name)
  return (
    <div className='nav-bar'>
      <p className = 'welcome'>Welcome, <span className = 'user'>{mappedName}</span></p>
      <NavLink className='nav' to='/'>Movies</NavLink>
      <NavLink className='nav' to='/favorites'><span>{favoriteLength}</span> Favorited</NavLink>
      <button
        className = 'logout'
        onClick = {() => this.logoutRedirect()}>log out</button>
    </div>
    )}
  }
}


export const mapStateToProps = (store) => {
  return {
    user: store.user,
    favorites: store.favorite
  }
}

export const mapDispatchToProps = (dispatch) => {
  return {
    logout: () => {
      dispatch(logout())
    },
     addMovieFunction: (getMovies) => {
    dispatch(addMovies(getMovies));
  },
    clearFavorites: () => {
      dispatch(clearFavorites());
    }
  }
}

NavBar.propTypes = {
addMovieFunction: PropTypes.func,
clearFavorites: PropTypes.func,
history: PropTypes.object,
location: PropTypes.object,
logout: PropTypes.func,
match: PropTypes.object,
user: PropTypes.array
}

export default connect(mapStateToProps, mapDispatchToProps)(NavBar)
