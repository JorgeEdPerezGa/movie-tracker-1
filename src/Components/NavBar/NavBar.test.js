import React from 'react';
import { shallow } from 'enzyme';
import NavBar from './NavBar';

let renderedNavBar;
beforeEach(() => {
   renderedNavBar = shallow(<NavBar />);
});
