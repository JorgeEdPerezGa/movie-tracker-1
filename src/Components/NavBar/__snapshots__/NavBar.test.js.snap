// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Test should match snapshot 1`] = `undefined`;
